tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -130,42 +142,178 @@ private static ArgumentParser argParser() {
                 .metavar("THROUGHPUT")
                 .help("throttle maximum message throughput to *approximately* THROUGHPUT messages/sec");
 
+        parser.addArgument("--num-threads")
+                .action(store())
+                .required(true)
+                .type(Integer.class)
+                .metavar("NUM-THREADS")
+                .dest("numThreads")
+                .help("The value bound of the random integers in message payload to simulate different compression ratio.");
+
+        parser.addArgument("--value-bound")
+                .action(store())
+                .required(true)
+                .type(Integer.class)
+                .metavar("VALUE-BOUND")
+                .dest("valueBound")
+                .help("The value bound of the random integers in message payload to simulate different compression ratio.");
+
         parser.addArgument("--producer-props")
                  .nargs("+")
                  .required(true)
                  .metavar("PROP-NAME=PROP-VALUE")
                  .type(String.class)
                  .dest("producerConfig")
-                 .help("kafka producer related configuaration properties like bootstrap.servers,client.id etc..");
+                 .help("kafka producer related configuration properties like bootstrap.servers,client.id etc..");
 
         return parser;
     }
 
+    private static class MetricStats {
+        private final List<Double> selectRateSamples;
+        private final List<Double> requestRateSamples;
+        private final Producer producer;
+        private final String clientId;
+        private final Map<String, String> tags;
+
+        MetricStats(Producer producer, String clientId) {
+            this.selectRateSamples = new ArrayList<Double>();
+            this.requestRateSamples = new ArrayList<Double>();
+            this.producer = producer;
+            this.clientId = clientId;
+            this.tags = new HashMap<String, String>();
+            this.tags.put("client-id", clientId);
+        }
+
+        public void recordMetricStats() {
+            Map<MetricName, Metric> metrics = producer.metrics();
+            selectRateSamples.add(metrics.get(new MetricName("select-rate", "producer-metrics", "", tags)).value());
+            requestRateSamples.add(metrics.get(new MetricName("request-rate", "producer-metrics", "", tags)).value());
+        }
+
+        public void printMetricStats() {
+            recordMetricStats();
+            Map<MetricName, Metric> metrics = producer.metrics();
+            double selectRateAvg = avg(selectRateSamples);
+            double requestRateAvg = avg(requestRateSamples);
+            double batchSizeAvg = metrics.get(new MetricName("batch-size-avg", "producer-metrics", "", tags)).value();
```

Comment:
Instead of `new MetricName("batch-size-avg", "producer-metrics", "", tags)`, I think we should use `metrics.metricName("batch-size-avg", "producer-metrics")`, right? The doc says `Please create MetricName by method {@link org.apache.kafka.common.metrics.Metrics#metricName(String, String, String, Map)}`.

Same for other MetricName instantiation.